[release/2.1] Include transitive refs in meta-package .nuspec files

dougbu opened the pull request:
- #30279
- ensure dependencies are up-to-date even when PatchConfig.props is empty
- also includes https://github.com/dotnet/aspnetcore/pull/30164/commits/5bafa2d364e6fd3e1bddfca0034b9a81b4e14db7 from #30164

dougbu committed:
Include transitive refs in meta-package .nuspec files
- #30279
- ensure dependencies are up-to-date even when PatchConfig.props is empty
- change only CoreFx and core-setup dependencies
- also change transitive references that weren't mentioned before

Pilchie commented:
Does this mean we'll publish a new version of the meta-packages for all patches, with their dependencies flattened?

I'm not against that (it makes things much better for people using 2.1 on .NET Framework for instance, but in the past there were some concerns about needing to update templates, and that affecting the offline package cache size in the SDK, etc.

dougbu commented:
@Pilchie our 2.1 builds always produce the two meta-packages. This will flatten only flatten dependencies from dotnet/CoreFx and dotnet/core-setup once it's ready for review. Basically, it'll be similar to the 3.1 and 5.0 special cases we have for some dotnet/runtime repos e.g. https://github.com/dotnet/aspnetcore/blob/release/5.0/eng/Versions.props#L173-L186

For now, this PR is
1. Not inclusive enough because it doesn't cover a few dotnet/CoreFx and dotnet/core-setup dependencies that are not yet referenced directly anywhere
2. Too inclusive because I updated a bunch of third-party dependencies that land in our meta-packages but are highly unlikely to get new versions we worry about. My preference here is to change their handling only if we need to react to security issues in those dependencies.

dougbu committed:
Correct and streamline internal builds
- correct manual internal builds
    - need to explicitly set `$(BuildNumber)` property in all build steps
    - rename `$(BuildScriptArgs)` to avoid circular reference
- skip tests by default in internal non-PR builds

nits:
- include new `$(BuildNumberArg)` in `$(SharedFxArgs)`
- run all test jobs in internal pull requests

dougbu commented:
Real test will be https://dev.azure.com/dnceng/internal/_build/results?buildId=1026812

dougbu commented:
Waiting for #30729 to work. Looks as if only ProdCon builds work internally for release/2.1☹️

dougbu committed:
Always skip tests in SharedFx and later jobs
- however test\SharedFx.UnitTests\SharedFx.UnitTests.csproj always runs as part of `BuildSharedFx` target

dougbu commented:
Do **not** merge pro-actively. Will rebase on #30729 commits to enable internal testing and validate that.

dougbu commented:
Confirmed this PR does what it says on the tin by comparing https://dev.azure.com/dnceng/internal/_build/results?buildId=1034483 artifacts with our 2.1.25 release. The Microsoft.AspNetCore.App.nuspec file now contains the transitive references from dotnet/CoreFx and dotnet/core-setup.

In last `push`, I removed #30729 fixes that matter only between branding and baseline updates.

wtgodbe commented on build/dependencies.props:
```diff
@@ -230,5 +230,10 @@
     <XunitRunnerVisualStudioPackageVersion>2.4.0</XunitRunnerVisualStudioPackageVersion>
+
+    <!-- Dependencies listed only to include otherwise-transitive references in Microsoft.AspNetCore.App.nuspec. -->
+    <MicrosoftDotNetPlatformAbstractionsPackageVersion>2.1.0</MicrosoftDotNetPlatformAbstractionsPackageVersion>
```
Are these all the latest?

dougbu replied:
No. The two below are out-of-date. Should I check Microsoft.Extensions.DependencyModel and all of the `System.*` packages above❔

wtgodbe replied:
Yeah, it'd be better to have the latest for everything

dougbu replied:
Agreed though I didn't check _everything_ in this file and instead focused on packages CoreFx, core-setup, extensions, and efcore

wtgodbe reviewed (approved):
Seems reasonable

dougbu committed:
Bump a few patch versions

dougbu commented on build/dependencies.props:
```diff
@@ -132,11 +132,11 @@
     <MicrosoftIdentityModelProtocolsWsFederationPackageVersion>5.2.0</MicrosoftIdentityModelProtocolsWsFederationPackageVersion>
-    <MicrosoftNETCoreApp10PackageVersion>1.0.15</MicrosoftNETCoreApp10PackageVersion>
-    <MicrosoftNETCoreApp11PackageVersion>1.1.12</MicrosoftNETCoreApp11PackageVersion>
+    <MicrosoftNETCoreApp10PackageVersion>1.0.16</MicrosoftNETCoreApp10PackageVersion>
+    <MicrosoftNETCoreApp11PackageVersion>1.1.13</MicrosoftNETCoreApp11PackageVersion>
```
These two version variables are only mentioned in external-dependencies.props but I don't want to mess with deleting them

dougbu commented on build/dependencies.props:
```diff
@@ -132,11 +132,11 @@
     <MicrosoftNETCoreApp20PackageVersion>2.0.9</MicrosoftNETCoreApp20PackageVersion>
     <MicrosoftNETCoreWindowsApiSetsPackageVersion>1.0.1</MicrosoftNETCoreWindowsApiSetsPackageVersion>
-    <MicrosoftNETTestSdkPackageVersion>15.9.0</MicrosoftNETTestSdkPackageVersion>
+    <MicrosoftNETTestSdkPackageVersion>15.9.2</MicrosoftNETTestSdkPackageVersion>
```
This and `$(SystemNetHttpWinHttpHandlerPackageVersion)` may only matter in tests. They just got caught in my NuGet checkification
